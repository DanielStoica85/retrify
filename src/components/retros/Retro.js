import React from 'react';
import {
    Card,
    CardBody,
    CardTitle,
    CardSubtitle,
    CardFooter
} from 'reactstrap';
import PropTypes from 'prop-types';

const Retro = props => {
    const { title, author, description, createdAt } = props.retro;
    return (
        <div>
            <Card className="mt-3">
                <CardBody>
                    <CardTitle>{title}</CardTitle>
                    <CardSubtitle> {description}</CardSubtitle>
                </CardBody>
                <CardFooter
                    style={{
                        backgroundColor: 'lightgray',
                        borderColor: 'lightgray'
                    }}
                >
                    Added by {author} on {createdAt}
                </CardFooter>
            </Card>
        </div>
    );
};

Retro.propTypes = {
    retro: PropTypes.object
};

export default Retro;
