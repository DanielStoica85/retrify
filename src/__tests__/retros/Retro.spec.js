import React from 'react';
import renderer from 'react-test-renderer';
import Retro from '../../components/retros/Retro';

describe('Retro component', () => {
    const props = {
        retro: {
            title: 'Test title',
            author: 'Test author',
            date: 'May 20, 2020'
        }
    };
    it('renders correctly', () => {
        const component = renderer.create(<Retro {...props} />).toJSON();
        expect(component).toMatchSnapshot();
    });
});
