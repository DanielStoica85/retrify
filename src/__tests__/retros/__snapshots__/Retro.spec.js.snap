// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Retro component renders correctly 1`] = `
<div>
  <div
    className="mt-3 card"
  >
    <div
      className="card-body"
    >
      <div
        className="card-title"
      >
        Test title
      </div>
      <div
        className="card-subtitle"
      >
        Added by 
        Test author
      </div>
    </div>
    <div
      className="card-footer"
      style={
        Object {
          "backgroundColor": "lightgray",
          "borderColor": "lightgray",
        }
      }
    >
      May 20, 2020
    </div>
  </div>
</div>
`;
